### PR TITLE
[#4] Implement IconTile component

### PR DESCRIPTION
**Pull request by TomomiKimura**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,6 +33,12 @@
 npm run dev
 ```
 
+#### ...fix "Port XXXX is already in use." error
+
+```sh
+./scripts/kill-servers.sh
+```
+
```
Comment (MiguelAraCo) [inline README.md:41]: We should move this and the script to a different PR to tackle it separately.

File: apps/web/pages/index.tsx [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
 export default function Web() {
-  return (
-    <div>
-      <h1>Web</h1>
-      <Button type="primary" onClick={() => console.log("I got clicked!")}>
-        Learn to Buy!
-      </Button>
-    </div>
-  );
+    return (
```
Comment (MiguelAraCo) [inline apps/web/pages/index.tsx:6]: Please revert the changes made to this file.

File: packages/ui/shared/IconTile/IconTile.module.css [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    font-weight: normal;
+}
+.description {
+    font-weight: normal;
+    padding: var(--space-xl);
+}
+/*.primary {*/
+/*    background-color: var(--colors-icontile-primary);*/
+/*    color: var(--colors-icontile-text);*/
+/*}*/
```
Comment (MiguelAraCo) [inline packages/ui/shared/IconTile/IconTile.module.css:30]: If this is not being used, please remove it.

File: packages/ui/shared/IconTile/IconTile.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    onClick,
+}) => {
+    return (
+        // DONE: Add condition that when there is only a description, show paragraph.
+        <button className={styles.iconTile} onClick={onClick}>
+            <div>
+                <div className={styles.logo} />
+                <div className={styles.titleContent}>
+                    <h2>{title}</h2>
+                    {description !== undefined ? description : null}
```
Comment (MiguelAraCo) [inline packages/ui/shared/IconTile/IconTile.tsx:24]: We may want to wrap this text around a `p` node and simplify the condition (`conditon && render block`)

File: packages/ui/shared/variables.css [hunk 5]
```diff
@@ -6,13 +6,18 @@
     --colors-secondary-text: #ffffff;
     --colors-error: #ff0000;
     --colors-error-text: #ffffff;
+    --colors-icontile-primary: #ffffff;
+    --colors-icontile-text: #414042;
+    --colors-icontile-primary-hover: #e7e5ea;
 
+    /* === Sizes === */
+    --size-icon-primary: 310px 230px;
+    --size-icon: 150px 150px;
```
Comment (MiguelAraCo) [inline packages/ui/shared/variables.css:15]: This seems to be specific to the icon tile component. We may want to move it to that file instead.

File: packages/ui/shared/IconTile/IconTile.tsx (hunk 4 above)
Comment (TomomiKimura) [reply packages/ui/shared/IconTile/IconTile.tsx]: I changed the line, thank you!

File: packages/ui/shared/IconTile/IconTile.module.css [hunk 6]
```diff
@@ -0,0 +1,28 @@
+.iconTile {
+  border: 0;
```
Comment (MiguelAraCo) [inline packages/ui/shared/IconTile/IconTile.module.css:2]: We probably don't need this line

File: packages/ui/shared/IconTile/IconTile.module.css [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  font-weight: bold;
+  text-transform: uppercase;
```
Comment (MiguelAraCo) [inline packages/ui/shared/IconTile/IconTile.module.css:4]: May be a good idea to group together the rules that are related

File: packages/ui/shared/IconTile/IconTile.module.css [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+.iconTile:hover {
+  /* TODO: change it so that it grows without altering the layout */
+  background-color: var(--colors-icontile-primary-hover);
+}
+.image {
+  margin: auto;
+  height: 100%;
+  width: 100%;
+  font-weight: normal;
+  font-size: 86px;
```
Comment (MiguelAraCo) [inline packages/ui/shared/IconTile/IconTile.module.css:23]: It may be better to assign a different class when displaying an icon. If we do that, we can place these rules there.

File: packages/ui/shared/IconTile/IconTile.module.css [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  height: 310px;
+  width: 230px;
+  padding: 20px;
+}
+.iconTile:hover {
+  /* TODO: change it so that it grows without altering the layout */
+  background-color: var(--colors-icontile-primary-hover);
+}
+.image {
+  margin: auto;
```
Comment (MiguelAraCo) [inline packages/ui/shared/IconTile/IconTile.module.css:19]: Is this to center the image? We can probably achieve the same using flexbox only

File: packages/ui/shared/IconTile/IconTile.module.css [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  height: 100%;
+  width: 100%;
+  font-weight: normal;
+  font-size: 86px;
+}
+.description {
+  font-weight: normal;
```
Comment (MiguelAraCo) [inline packages/ui/shared/IconTile/IconTile.module.css:26]: This shouldn't be needed since the default should be normal.

File: packages/ui/shared/IconTile/IconTile.module.css [hunk 11]
```diff
@@ -0,0 +1,28 @@
+.iconTile {
+  border: 0;
+  font-weight: bold;
```
Comment (MiguelAraCo) [inline packages/ui/shared/IconTile/IconTile.module.css:3]: We may want to put text rules directly on the element that uses them so that non-related elements don't inherit them.

File: packages/ui/shared/IconTile/IconTile.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+export default {
+  component: IconTile,
+} as ComponentMeta<typeof IconTile>;
+
+export const WithImageAndDescription: ComponentStoryObj<typeof IconTile> = {
+  args: {
+    title: "equipment",
+    description: "Browse 200+ Systems in Stock",
+    image: {
+      src: "https://via.placeholder.com/125x125/0000FF",
```
Comment (MiguelAraCo) [inline packages/ui/shared/IconTile/IconTile.stories.tsx:14]: What would happen if the image is really big? Or really small?

File: packages/ui/shared/IconTile/IconTile.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+      alt: "Logo of the equipment",
+    },
+  },
+};
+
+export const WithIconAndDescription: ComponentStoryObj<typeof IconTile> = {
+  args: {
+    image: "bi bi-briefcase",
```
Comment (MiguelAraCo) [inline packages/ui/shared/IconTile/IconTile.stories.tsx:32]: This should only be the icon name, shouldn't it?

File: packages/ui/shared/IconTile/IconTile.tsx [hunk 14]
```diff
@@ -0,0 +1,40 @@
+import * as React from "react";
+import { ReactNode } from "react";
+import { Icon, IconProps } from "../Icon/Icon";
+
+import styles from "./IconTile.module.css";
+
+export type IconTileProps = {
+  title: string;
+  description?: string;
+  image?:
```
Comment (MiguelAraCo) [inline packages/ui/shared/IconTile/IconTile.tsx:10]: May be good to add some documentation to make it clear that it can either be an image or a Bootstrap icon

File: packages/ui/shared/IconTile/bootstrap.module.css [hunk 15]
```diff
@@ -0,0 +1,69 @@
+:root {
```
Comment (MiguelAraCo) [inline packages/ui/shared/IconTile/bootstrap.module.css:1]: Not sure what this file is for. Let's remove it if it isn't being used.

File: packages/ui/shared/variables.css [hunk 16]
```diff
@@ -19,6 +19,10 @@
   --colors-disabled: #e5e6e7;
   --colors-disabled-text: #9ea0a1;
 
+  --colors-icontile-primary: #ffffff;
```
Comment (MiguelAraCo) [inline packages/ui/shared/variables.css:22]: Reminder that at some point we should move component-specific variables to a different file.

File: packages/ui/shared/variables.css [hunk 17]
```diff
@@ -39,5 +43,6 @@
   --font-h3: 28px;
   --font-h4: 24px;
   --font-h5: 18px;
+  --font-newh5: 16px;
```
Comment (MiguelAraCo) [inline packages/ui/shared/variables.css:46]: We need to introduce a proper design system to avoid names like this one. For the time being, let's add a TODO to rename this since the name isn't clear.

**Review by MiguelAraCo — COMMENTED** [review]
Left a couple of comments 🙇

File: packages/ui/shared/IconTile/IconTile.module.css (hunk 6 above)
Comment (TomomiKimura) [reply packages/ui/shared/IconTile/IconTile.module.css:2]: 👍

File: packages/ui/shared/IconTile/IconTile.module.css (hunk 11 above)
Comment (TomomiKimura) [reply packages/ui/shared/IconTile/IconTile.module.css:3]: 👍

File: packages/ui/shared/IconTile/IconTile.module.css (hunk 7 above)
Comment (TomomiKimura) [reply packages/ui/shared/IconTile/IconTile.module.css:4]: 👍

File: packages/ui/shared/IconTile/IconTile.module.css (hunk 9 above)
Comment (TomomiKimura) [reply packages/ui/shared/IconTile/IconTile.module.css:19]: 👍

File: packages/ui/shared/IconTile/IconTile.module.css (hunk 8 above)
Comment (TomomiKimura) [reply packages/ui/shared/IconTile/IconTile.module.css:23]: 👍

File: packages/ui/shared/IconTile/IconTile.module.css (hunk 10 above)
Comment (TomomiKimura) [reply packages/ui/shared/IconTile/IconTile.module.css:26]: Had a `font-weight` on a parent element. Removed it and removed this as well 👍

File: packages/ui/shared/IconTile/IconTile.stories.tsx (hunk 12 above)
Comment (TomomiKimura) [reply packages/ui/shared/IconTile/IconTile.stories.tsx:14]: I added stories to test this, and right now it is squishing the images in an undesired way. I'll create a ticket to track this and fix it later 🙇‍♀️

Comment (TomomiKimura) [reply packages/ui/shared/IconTile/IconTile.stories.tsx:14]: Here's the ticket: https://github.com/TomomiKimura/medical-equipment/issues/14

File: packages/ui/shared/IconTile/bootstrap.module.css (hunk 15 above)
Comment (TomomiKimura) [reply packages/ui/shared/IconTile/bootstrap.module.css:1]: Oh I didn't notice that I left it behind. I'll remove it 👍

File: packages/ui/shared/variables.css (hunk 16 above)
Comment (TomomiKimura) [reply packages/ui/shared/variables.css:22]: Agree! I'll create a ticket so that we don't forget.

File: packages/ui/shared/variables.css (hunk 17 above)
Comment (TomomiKimura) [reply packages/ui/shared/variables.css:46]: Agree. I'm not using this variable anymore, but I added a TODO to introduce design tokens. I'll create a ticket for that as well 🙇‍♀️.

**Review by MiguelAraCo — APPROVED** [review]
LGTM!
